### PR TITLE
Added User registration without email

### LINKED ISSUE #188: Allow new users without an email address
It appears it only allows you to 'invite' users by their email addresses.
Not good when its children in a house, who have no email addresses.
You have to create a bogus email, then hand edit the username and password in the mysql tables directly.
Very poor.

### PR DESCRIPTION
**Pull request by lclc98**
Adds the ability to register new users without the need send them an invite.
This should resolve #188

### CODE REVIEW

File: grails-app/controllers/streama/UserController.groovy [hunk 1]
```diff
@@ -119,6 +122,37 @@ class UserController {
     respond userInstance, [status: CREATED]
   }
 
+  @Transactional
+  def saveAndCreateUser() {
```
Comment (gkiko) [inline grails-app/controllers/streama/UserController.groovy]: 1. This method is very much like the `saveAndInviteUser`. Code duplication should be avoided([DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself)).
2. And it does more than one thing([SRP](https://en.wikipedia.org/wiki/Single_responsibility_principle)). Its name says that.

I think that you can break both `saveAndInviteUser ` and `saveAndCreateUser` methods into smaller pieces while eliminating duplicated lines. Write separate methods that _creates and validates the user_, _saves the user and role in database_ and _sends email_.

This pseudocode will give you an idea on how to start
```
def createAccountWithEmail() {
  User u = createUser()
  if (u==null) {
    return
  }
  createRole(u)
  sendEmail(u)
  u.save
}
``` 

@dularion what are your thought on this?

File: grails-app/assets/javascripts/streama/controllers/modal-user-ctrl.js [hunk 2]
```diff
@@ -63,4 +72,20 @@ angular.module('streama').controller('modalUserCtrl', [
 				});
 		};
 
+    $scope.saveAndCreateUser = function (user) {
```
Comment (gkiko) [inline grails-app/assets/javascripts/streama/controllers/modal-user-ctrl.js]: Duplicate of `saveAndInviteUser`

File: grails-app/assets/javascripts/streama/controllers/settings-users-ctrl.js [hunk 3]
```diff
@@ -9,8 +9,44 @@ angular.module('streama').controller('settingsUsersCtrl', ['$scope', 'apiService
 
 	});
 
-	$scope.openUserModal = function (user) {
-		modalService.userModal(user, function (data) {
+  $scope.openUserEditModal = function (user) {
+    modalService.openUserEditModal(user, function (data) {
```
Comment (gkiko) [inline grails-app/assets/javascripts/streama/controllers/settings-users-ctrl.js]: You can extract anonymous function to eliminate duplication

File: grails-app/controllers/streama/UserController.groovy [hunk 4]
```diff
@@ -59,10 +61,11 @@ class UserController {
 
   def checkAvailability() {
     def username = params.username
+    def isInvite = params.isInvite
     def result = [:]
 
     if (User.findByUsername(username)) {
-      result.error = "User with that E-Mail-Address already exists."
+      result.error = (isInvite == "true") ? "User with that E-Mail-Address already exists." : "Username already exists."
```
Comment (gkiko) [inline grails-app/controllers/streama/UserController.groovy]: I think server shouldn't really care about customized error messages.
You can only pass username and to check against database records. Send *HEAD* or *GET* request and see if user does not exists return *404*, if user is in db return *200*. Based on server response you can show relevant message on client side. Besides flag argument will be removed in `checkAvailability` function 👍

**Comment by lclc98** [comment]
@gkiko I have actually never worked with this project language and structure before, so I am still learning the basics, but I will look into the what you said so that I can make my future PR better.

**Pull request merged**